src/fabric.c

```diff
@@ -214,22 +214,19 @@ static int ofi_register_provider(struct fi_provider *provider, void *dlhandle)
 	ctx = (struct fi_prov_context *) &provider->context;
 	ctx->is_util_prov = (ofi_util_name(provider->name, &len) != NULL);
 
-	/* Util providers are never filtered, as they cannot be used
-	 * by themselves.
-	 */
-	if (!ctx->is_util_prov) {
-		if (ofi_apply_filter(&prov_filter, provider->name)) {
-			FI_INFO(&core_prov, FI_LOG_CORE,
-				"\"%s\" filtered by provider include/exclude "
-				"list, skipping\n", provider->name);
-			ret = -FI_ENODEV;
-			goto cleanup;
-		}
-
-		if (ofi_apply_filter(&prov_log_filter, provider->name))
-			ctx->disable_logging = 1;
+	/* This would allow util provider filtering as well. User should take
+	 * care to provide a core provider when filtering util provider */
+	if (ofi_apply_filter(&prov_filter, provider->name)) {
```

Comment:
The original code allowed for a util provider to still be used with a core provider when only the core provider is specified as part of the filter.  E.g. FI_PROVIDER=verbs would allow RxM to layer over verbs in order to support RDM EPs.  This change now requires the user to set FI_PROVIDER=verbs,ofi_rxm.  This means that the use of the util provider is no longer 'hidden'.

If the intent is to allow disabling a util provider, then we need to enhance the check somehow to only apply the filter if it has been negated, rather than a positive check.  Hopefully that makes sense.

Reply:
Btw - I will make this change and push up a new PR for this.

Reply:
> If the intent is to allow disabling a util provider, then we need to enhance the check somehow to only apply the filter if it has been negated, rather than a positive check. 

Yes, this is the intent. Thanks!